torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -63,17 +63,17 @@ void propagateNode(Node* n) {
   auto graph = n->owningGraph();
   WithInsertPoint guard(n);
   for (size_t i = 0; i < outputs.size(); ++i) {
-    try {
-      auto new_output = graph->insertConstant(outputs[i]);
+
+    auto new_output = tryInsertConstant(*graph, outputs[i]);
+    if (new_output) {
+      // auto new_output = tryInsertConstant(graph, outputs[i]);
       if (outputs[i].isNone()) {
-        new_output->setType(n->outputs()[i]->type());
+        (*new_output)->setType(n->outputs()[i]->type());
```

Comment:
Can you not do new_output->setType... ? would think pointer works here but maybe not. Also, semi related if you don't mind adding a comment above `throw c10::Error("Can't insert requires grad as ` to say something along the lines of `// error gets caught within constant prop`